src/mono/mono/eglib/giconv.c

```diff
@@ -282,7 +282,7 @@ decode_utf8 (char *inbuf, size_t inleft, gunichar *outchar)
 {
 	unsigned char *inptr = (unsigned char *) inbuf;
 	gunichar u;
-	int n, i;
+	guint n;
```

Comment:
`size_t` is `unsigned int` eventually. I don't think that it is necessary to change the type of `n` to `size_t`? Based on the implementation, it is known that `n` would be much smaller than INT_MAX. I don't think that it is necessary to do the cast on `n` when return.

Reply:
size_t depens on size of architecture so it will be 4 or 8 bytes. It is always better to align types as much as possible, and since you do the change I don't see why we shouldn't align to compatible types.